src/vaadin-date-time-picker.html

```diff
@@ -21,15 +32,38 @@
       :host([hidden]) {
         display: none !important;
       }
+
+      .slot-container {
+        display: flex;
+      }
+
+      [part="date"],
+      .slot-container ::slotted([slot="date-picker"]) {
+        min-width: 0;
+        flex: 1 1 auto;
+      }
+
+      [part="time"],
+      .slot-container ::slotted([slot="time-picker"]) {
+        min-width: 0;
+        flex: 1 1.65 auto;
+      }
     </style>
-    <vaadin-custom-field id="customField" on-value-changed="__customFieldValueChanged" i18n="[[__customFieldValueFormat]]" theme$="[[theme]]">
-      <slot name="date-picker">
-        <vaadin-date-picker part="date" theme$="[[theme]]"></vaadin-date-picker>
-      </slot>
-      <slot name="time-picker">
-        <vaadin-time-picker part="time" theme$="[[theme]]"></vaadin-time-picker>
-      </slot>
-    </vaadin-custom-field>
+    <vaadin-date-time-picker-custom-field
+        id="customField"
+        on-value-changed="__customFieldValueChanged"
+        i18n="[[__customFieldValueFormat]]"
+        theme$="[[theme]]"
+      >
+      <div class="slot-container">
```

Comment:
The only change in this diff block is adding this `slot-container` wrapper and some indentation and formatting.

This wrapper element is needed to be able to style the slotted elements with Shady CSS compliant selectors (so it works in Edge and IE11 too). See https://github.com/webcomponents/polyfills/tree/master/packages/shadycss#selector-scoping

Reply:
Wouldn't `:host ::slotted([slot="date-picker"])` work (so the extra wrapper wouldn't be needed)?

Reply:
I tried removing the slot-container element and using `:host ::slotted([slot="date-picker"])` or `vaadin-date-time-picker-custom-field ::slotted([slot="date-picker"])` instead but I couldn't get it to work in shady DOM. I also had to move the display flex to:
```css
.inputs-wrapper {
    display: flex;
}
```
in the `theme-for="vaadin-date-time-picker-custom-field"` like it was in a previous commit.

The behaviour in `test/visual/flex-behaviour.html` was incorrect in Edge and IE11 (shady DOM) in this case for the "Full width (slotted)" visual test.

I think the problem is shady DOM scoping mechanism. With the extra slot container element as a light child of the custom field it works because then everything in the selector for the slotted element is in the same shady scope. I suspect the `:host` is just not supported by shady CSS in this context as a parent selector for slotted.

So to me it looks like this slot-container wrapper is indeed necessary to make this work consistently in all cases. It can be simplified after we drop support for IE11 and when Edge moves to Chromium (no need to support shady DOM anymore then).